cpp/include/cudf/detail/utilities/device_operators.cuh

```diff
@@ -73,6 +73,14 @@ struct DeviceCount {
   }
 };
 
+/* @brief `row_number` operator - used in rolling windows */
+struct DeviceRowNumber {
+  template <typename T>
+  static constexpr T identity() {
+    return T{0};
+  }
```

Comment:
We shouldn't have to define an operator for this. Just to get the identity value of 0.

Reply:
> It seems that process_rolling_window can generate it on its own since it's specialized anyway...
Would you kindly make this change as a part of this PR?

I tried refactoring things to get this to work. Most of the `identity()` calls can be moved to the `process_rolling_window()` overloads, except for `__host__ DeviceMin/DeviceMax::identity()`, because `process_rolling_window()` is a `__device__` function. 

I'm not currently equipped to rewrite `identity()` as a `__device__` function, since I don't know if there's an appropriate equivalent of `cudaGetSymbolAddress()`. :/ I suspect that's why the identity object is being constructed and passed down the chain.